cpp/src/utilities/column_utils.hpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file Small utility functions for working with @ref gdf_column structures,
+ * which are raw C structures without any guarantees and operators defined for
+ * them, and thus require a bit of care when handling. This care and due
+ * diligence tends to be rather repetitive; hence this file.
+ *
+ * @note since the "right" way to work with these structures is pass references,
+ * but the "C" way of doing the same involves pointers, many functions are
+ * implemented for both.
+ */
+#ifndef UTILITIES_COLUMN_UTILS_HPP_
+#define UTILITIES_COLUMN_UTILS_HPP_
+
+#include <utilities/type_dispatcher.hpp>
+
+#include <cudf.h>
+
+#include <algorithm>
+
+namespace cudf {
+
+template<typename T>
+T* get_data(const gdf_column& column) noexcept { return static_cast<T*>(column.data); }
+
+template<typename T>
+T* get_data(const gdf_column* column) noexcept { return get_data<T>(*column); }
+
+constexpr inline bool is_an_integer(gdf_dtype element_type)
```

Comment:
I'd prefer that this kind of thing go through the `type_dispatcher` to limit the amount of code that need change when new types are added/modified, e.g., if/when we add unsigned integral types.

Something like this should do it:

```
struct integral_check{
   template <typename T>
   constexpr bool operator()(void) const noexcept{
      return std::is_integral<T>::value;
   }
};

...

return cudf::type_dispatcher(element_type, integral_check{});
```

Reply:
Why would you prefer using a complex and not-narrattively flowing gadget for this kind of a function? As for using `std::is_integral` - we could use that, but then I'd need the inverse function of cudf::size_of (i.e. `cudf::type_of`).

Actually, a `cudf::type_of` might be nice, because we could use it in the implementation type_dispatcher as well.

Also, using `std::integral` requires that we guarantee that only GDF_INTXX or GDF_UINTXX types will use an unwrapped integer type, and vice versa, that no type which "is an integer" will ever be wrapped.

If you're willing to make that guarantee, I'll switch to your suggestion or use `cudf::type_of` if you approve of it.

Reply:
I think I wasn't clear.

I'm not suggesting getting rid of `is_an_integer`. Instead, my suggestion was to update the implementation of `is_an_integer`. 

For example:

```
struct integral_check{
   template <typename T>
   constexpr bool operator()(void) const noexcept{
      return std::is_integral<T>::value;
   }
};
constexpr inline bool is_an_integer(gdf_dtype element_type){
   return cudf::type_dispatcher(element_type, integral_check{});
}
```

Reply:
You were clear... I'm just questioning whether it's an upgrade. It's more convoluted and obscure (as much as I like our type dispatcher), and requires making the implicit assumption I mentioned.

Reply:
> Why would you prefer using a complex and not-narrattively flowing gadget for this kind of a function?

Because then when we add `GDF_UINT*` I have to remember to go back and update this function. Ideally, I want to keep all of the changes required when a new type is added or changed to be limited to as few places as possible. 

> cudf::type_of

Making sure I understand, you're suggesting a trait that maps `gdf_dtype` to concrete C++ types? If so, I support that and updating `type_dispatcher` to use the trait.

That said, I don't know how that'd help the implementation of `is_an_integer`, you'd still need a switch statement over `element_type`. That's exactly the kind of thing I'm trying to minimize. 

> Also, using std::integral requires that we guarantee that only GDF_INTXX or GDF_UINTXX types will use an unwrapped integer type, and vice versa, that no type which "is an integer" will ever be wrapped.

I feel comfortable making that guarantee for now. If we _do_ need to use a wrapper instead, we can always overload `std::is_integral<T>` for user-defined types.

Reply:
Given your guarantee, let's not split hairs and I'll go with your suggestion.